athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/pushdown/LimitPushdownSubType.java

```diff
@@ -19,22 +17,23 @@
  * limitations under the License.
  * #L%
  */
+package com.amazonaws.athena.connector.lambda.metadata.optimizations.pushdown;
 
-public enum SplitProperties
+public enum LimitPushdownSubType
+        implements PushdownSubTypes
 {
-    LOCATION(ExampleMetadataHandler.PARTITION_LOCATION),
-    SERDE(ExampleMetadataHandler.SERDE),
-    SPLIT_PART("SPLIT_PART");
+    ALL("all");
```

Comment:
should we give it a better name than all?

Reply:
we were between "ALL" and "FULL", and will probably add "PARTIAL" in the future. So between ALL and FULL we picked ALL.

Reply:
Consulted with @burhan94 , we can leave this as ALL because there's no other types of limits from the perspective of the connector. In the engine, the engine will handle the logic of deciding if it's pushing down a full or a partial limit, but the connector does not need to care.

Reply:
After talking more to @henrymai , he brought up possibilities of pushing down things like functions or variables with the LIMIT, and how non-integers are possible (already possible in Impala and Postgres, as examples). So I've refactored this to be "INTEGER_CONSTANT".